File: test/support/distances.py
from khiva.array import *
from khiva.distances import *
from khiva.library import *

set_backend(KHIVABackend.KHIVA_BACKEND_CPU)
set_device(0)

a = Array.from_list([[1, 1, 1, 1, 1], [2, 2, 2, 2, 2], [3, 3, 3, 3, 3]], dtype.s64)

print('dtw')
print(dtw(a).to_list())
print()

print('euclidean')
print(euclidean(a).to_list())
print()

print('hamming')
print(hamming(a).to_list())
print()

print('manhattan')
print(manhattan(a).to_list())
print()

print('sbd')
print(sbd(a).to_list())
print()

print('squared_euclidean')
print(squared_euclidean(a).to_list())
print()

File: lib/khiva/distances.rb
module Khiva
  module Distances
    class << self
      def dtw(tss)
        result = Utils.create_ptr
        FFI.call(:dtw, tss, result)
        Array.new(result)
      end

      def euclidean(tss)
        result = Utils.create_ptr
        FFI.call(:euclidean, tss, result)
        Array.new(result)
      end

      def hamming(tss)
        result = Utils.create_ptr
        FFI.call(:hamming, tss, result)
        Array.new(result)
      end

      def manhattan(tss)
        result = Utils.create_ptr
        FFI.call(:manhattan, tss, result)
        Array.new(result)
      end

      def sbd(tss)
        result = Utils.create_ptr
        FFI.call(:sbd, tss, result)
        Array.new(result)
      end

      def squared_euclidean(tss)
        result = Utils.create_ptr
        FFI.call(:squared_euclidean, tss, result)
        Array.new(result)
      end
    end
  end
end
